xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.cpp

```diff
@@ -25,13 +25,13 @@ using namespace AE;
 using namespace ActiveAE;
 using namespace std::chrono_literals;
 
-CActiveAESink::CActiveAESink(CEvent *inMsgEvent) :
-  CThread("AESink"),
-  m_controlPort("SinkControlPort", inMsgEvent, &m_outMsgEvent),
-  m_dataPort("SinkDataPort", inMsgEvent, &m_outMsgEvent)
+CActiveAESink::CActiveAESink(CEvent* inMsgEvent)
+  : CThread("AESink"),
+    m_controlPort("SinkControlPort", inMsgEvent, &m_outMsgEvent),
+    m_dataPort("SinkDataPort", inMsgEvent, &m_outMsgEvent),
+    m_sink(nullptr)
```

Comment:
Why is only `m_sink` moved to the initializer list? This kind selective change is done in some more instances of this PR and I don't get the reasoning.

Also default member initialization is preferred over initializer list anyway.

Reply:
This PR fixes the issues reported by cppcheck, as you can see in the title and commit messages.

Reply:
Yeah, I got that 🙂 But I still think that we shouldn't make incomplete changes just for the sake of ticking a box in a tool.

Reply:
The argument for going "above and beyond" is a strong one, but where do you stop? Porting the entire codebase to initializer lists?

As we do just fine without the changes recognized by cppcheck, I think there's value in just silencing warnings to better discover the errors later that actually matter.